src/lib.rs

```diff
@@ -106,6 +106,9 @@ pub enum Error {
         actual: ShapeType,
     },
     InvalidShapeRecordSize,
+    /// The Polygon shape read from the file cannot be converted to a
+    /// destination type (e.g. geo-traits types).
+    IncompatiblePolygonType(String),
```

Comment:
I think I would prefer a dedicated variant like `OrphanedInnerRing` or `MissingOuterRing`  or even `InnerRingWithoutOuter`  since that's the only error returned by the try_from